src/parser/xasl_generation.c

```diff
@@ -25217,6 +25204,36 @@ pt_to_merge_insert_query (PARSER_CONTEXT * parser, PT_NODE * select_list, PT_MER
   return subq;
 
 error_exit:
+
+  if (subq)
+    {
+      parser_free_tree (parser, subq);
+    }
+
+  if (corr_subq)
+    {
+      parser_free_tree (parser, corr_subq);
+    }
```

Comment:
If you are coming from line 25149, you may experience problems with free corr_subq and attempting free again in expr.
Please refer to the comments on line 25197 of the code before the change

Reply:
Ok, you are right. I will fix it.